test/misc.jl

```diff
@@ -27,6 +27,8 @@ for backend in [
     ADTypes.AutoDiffractor(),
     ADTypes.AutoEnzyme(),
     ADTypes.AutoEnzyme(mode = :forward),
+    ADTypes.AutoEnzyme(function_annotation = Val{:forward}),
+    ADTypes.AutoEnzyme(mode = :reverse, function_annotation = Val{:duplicated}),
```

Comment:
this isn't supported by the spec above (which requires annotation, not a symbol), no?

Reply:
Yeah this is just for testing printing, I can switch

Reply:
Actually I'd like to keep it this way for easy visual check. The annotation from EnzymeCore doesn't change much but if I also provide a realistic `mode` argument the print can get rather long and it's less obvious whether there's e.g. a comma mistake

Reply:
I get that, but on the other hand I'm worried that someone will see this, and assume that this is valid/the way it's supposed to be specified. And per the docstring, this is explicitly not supported.

Reply:
Exactly, so why would they look at the test and not trust the docstring?

Reply:
I can add a comment above the test saying "DONT DO THIS, THIS IS BAD, LOOK AD THE DOCSTRING FOR PROPER KWARG DOCTRINE."

Reply:
I guess I could be okay with such a comment, but isn't just using it properly easier?

Reply:
Then we'd have to use it properly for every other backend. Bring in FiniteDifferences to have a proper fdm object for AutoFiniteDifferences, give a proper RuleConfig to AutoChainRules, etc.
This is just for visual checking of the display code and it actually helps to keep things concise for our eyes to assess the result, so I really don't think we need to worry that much.

Reply:
Okay, then I'm fine with this [but a comment warning users would be appreciated -- since I know a bunch of folks that just copy code from tests and try to use]

Reply:
Right now the printout from the test is the following, each backend fits nicely on one line and we see what's missing without scrolling like a madman:

```julia
AutoChainRules(ruleconfig=:rc)
AutoDiffractor()
AutoEnzyme()
AutoEnzyme(mode=:forward)
AutoEnzyme(function_annotation=Val{:forward})
AutoEnzyme(mode=:reverse, function_annotation=Val{:duplicated})
AutoFastDifferentiation()
AutoFiniteDiff()
AutoFiniteDiff(fdtype=:fd, fdjtype=:fdj, fdhtype=:fdh)
AutoFiniteDifferences(fdm=:fdm)
AutoForwardDiff()
AutoForwardDiff(chunksize=3, tag=:tag)
AutoPolyesterForwardDiff()
AutoPolyesterForwardDiff(chunksize=3, tag=:tag)
AutoReverseDiff()
AutoReverseDiff(compile=true)
AutoSymbolics()
AutoTapir()
AutoTapir(safe_mode=false)
AutoTracker()
AutoZygote()
AutoSparse(dense_ad=AutoForwardDiff())
AutoSparse(dense_ad=AutoForwardDiff(), sparsity_detector=FakeSparsityDetector(), coloring_algorithm=FakeColoringAlgorithm())
```

Reply:
I have added this warning:
```
#=
The following tests are only for visual assessment of the printing behavior.
They do not correspond to proper use of ADTypes constructors.
Please refer to the docstrings for that.
=#
```